apps/app/src/app/shared/layout/page-nav/page-nav.component.ts

```diff
@@ -15,17 +34,42 @@ import { pageNavTmpl } from '~/app/shared/layout/page-nav/page-nav-outlet.compon
 				<div class="space-y-2 px-1">
 					<h3 class="font-medium">On this page</h3>
 					<ul class="m-0 flex list-none flex-col">
-						<ng-content />
+						@for (link of links(); track link.id) {
+							<spartan-page-nav-link [ngClass]="{ 'pl-4': link.isNested }" [fragment]="link.id" [label]="link.label" />
+						} @empty {
+							@if (isDevMode()) {
+								[DEV] Nothing to see here!
+							}
+						}
 					</ul>
 				</div>
 			</hlm-scroll-area>
 		</ng-template>
 	`,
 })
-export class PageNavComponent implements AfterViewInit, OnDestroy {
+export class PageNavComponent implements OnInit, AfterViewInit, OnDestroy {
 	@ViewChild('pageNav', { static: true })
 	pageNavTpl?: TemplateRef<unknown>;
 
+	isDevMode = signal(isDevMode());
+	links = signal<SamePageAnchorLink[]>([]);
+
+	private page: HTMLElement = (inject(ElementRef).nativeElement as HTMLElement).previousSibling as HTMLElement;
```

Comment:
We might want to be a little more careful here. I'd probably make this an optional `HTMLElement` and also add a check that we are only executing the logic in ngOnInit if the platform is browser. Thoughts?

Reply:
I completely forgot about the SSR aspect and only considered SSG. Analog is pretty much my first "proper" experience with SSR, so thanks for pointing that out!

What would you expect to happen when this component is used in an SSR context? This component is seemingly useless outside SPA/SSR, so I could update this component's constructor with this:
```ts
import { isDevMode, PLATFORM_ID } from '@angular/core';
import { isPlatformServer } from '@angular/common';
//...
private readonly platformId = inject(PLATFORM_ID);

constructor() {
  if (isDevMode() && isPlatformServer(this.platformId)) {
    throw new Error("This component should not be used for non-SSG/SPA pages.");
  }
}
```

That way, we don't need to worry about this running on the server.

Reply:
I don't think that we need to throw an error. I'd just return early, but maybe leave your explanation in a comment.

Reply:
Sure! Also, I think I should swap `ElementRef` with `Renderer2` as the latter would not break in the SSR context.

Reply:
If you want to make that change I am cool with it. However, I think the early return on SSR would be enough? LMK what you think

Reply:
I've added a commit with the change for an early return. Renderer2 doesn't seem relevant here once I made the change so I've skipped that for now.

Added a comment for the ElementRef line, and a console error for when it runs on the server.